daemon-tests/tests/rollover.rs

```diff
@@ -235,51 +283,108 @@ async fn retry_rollover_an_open_cfd() {
         )
         .await;
 
-    // We mock the initial oracle event-id for the rollover retry again
-    mock_oracle_announcements(
+    // The expected fee is less within the test setup, because we now charge only one full turn,
+    // i.e. 24h
+    rollover(
         &mut maker,
         &mut taker,
-        contract_setup_oracle_data_announcement.clone(),
+        order_id,
+        OliviaData::example_0(),
+        Some((
+            taker_commit_txid_after_contract_setup,
+            OliviaData::example_0().announcement().id,
+        )),
+        predict_fees.calculate_for_hours(24),
     )
     .await;
 
-    // We expect that the rollover retry won't add additional costs
-    let (expected_maker_fee, expected_taker_fee) = fee_structure.predict_fees(24);
+    // 3. Ensure that we can do another rollover after the retry
+    // After another rollover we expect to be charged for 48h
+    rollover(
+        &mut maker,
+        &mut taker,
+        order_id,
+        OliviaData::example_1(),
+        None,
+        predict_fees.calculate_for_hours(48),
+    )
+    .await;
+}
+
+#[otel_test]
+async fn retry_rollover_an_open_cfd_from_previous_rollover() {
```

Comment:
🙃 And this one could be `given_maker_one_rollover_ahead_when_retry_then_taker_catches_up`.

Reply:
:wrench: We could order the tests by complexity of the scenario:

1. Taker behind by one rollover.
2. Taker behind by two rollovers.
3. Taker failed the very first rollover.